python/ray/rllib/env/vector_env.py

```diff
@@ -127,15 +145,36 @@ def __init__(self,
                  num_envs,
                  action_space=None,
                  observation_space=None):
-        _VectorizedGymEnv.__init__(self, make_env, [], num_envs, action_space,
-                                   observation_space)
+        self.make_local_env = make_env
+        self.num_envs = num_envs
+        self.initialized = False
+        logger.warning(
+            "Creating throwaway env to get action and obs space. To avoid "
+            "resource overheads, your env should defer any expensive "
+            "initialization to reset().")
+        dummy = make_env(0)
```

Comment:
You shouldn't need dummy env as you should have action_space / observation_space provided from dummy env created in policy evaluator constructor.

Reply:
and if creating dummy env, calling dummy.close() at the end might be a good idea?

Reply:
Right, fixed.